cni-plugin/linkerd-cni-validator/src/main.rs

```diff
@@ -118,52 +133,53 @@ async fn validate_outbound_redirect(
         }
     };
 
-    let mut buf = [0u8; 100];
-    let read_sz = stream.read(&mut buf[..REDIRECT_RESPONSE.len()]).await?;
-    let resp = String::from_utf8(buf[..REDIRECT_RESPONSE.len()].to_vec())?;
+    let mut buf = [0u8; 8];
+    let read_sz = stream.read_exact(&mut buf).await?;
+    let resp = String::from_utf8(buf[..expected_resp.len()].to_vec())?;
     debug!(redirect_response = %resp, bytes_read = %read_sz);
     anyhow::ensure!(
-        resp == REDIRECT_RESPONSE,
+        resp == expected_resp,
         "expected client to receive {:?}, got {:?} instead",
-        REDIRECT_RESPONSE,
+        expected_resp,
         resp
     );
     Ok(())
 }
 
-#[tracing::instrument(name = "outbound_server", skip(ready_tx, shutdown))]
+#[tracing::instrument(name = "outbound_server", skip(ready, shutdown))]
 async fn outbound_serve(
     listen_addr: SocketAddr,
-    ready_tx: oneshot::Sender<Result<()>>,
+    ready: Arc<Notify>,
     shutdown: kubert::shutdown::Watch,
+    resp: String,
 ) -> Result<()> {
     let listener = TcpListener::bind(listen_addr)
         .await
         .expect("Failed to bind server");
     info!("Listening for incoming connections");
 
-    if ready_tx.send(Ok(())).is_err() {
-        error!("Failed to send 'ready' signal, receiver dropped");
-        anyhow::bail!("Failed to bind server");
-    }
+    ready.notify_one();
 
-    let resp_bytes = REDIRECT_RESPONSE.as_bytes();
-    tokio::spawn(accept(listener, resp_bytes));
+    let accept_task = tokio::spawn(accept(listener, resp)).in_current_span();
```

Comment:
Oh, I see. The span should be on the spawned future, not the handle.
```suggestion
    let accept_task = tokio::spawn(accept(listener, resp).in_current_span());
```

Reply:
Uf, yeah, I was wondering why my spans weren't showing... I noticed I had another span on the handle instead of the future, I've changed both. Thanks, this was super useful!